instrumentation/opentelemetry_ex_aws/test/opentelemetry_ex_aws/dynamo_db_test.exs

```diff
@@ -0,0 +1,546 @@
+               expression_attribute_values: [partition_key: "key1"],
+               scan_index_forward: true,
+               select: :all_attributes,
+               limit: 2
+             )
+             |> ExAws.request()
+
+    assert [
+             %{"attribute_name" => "value1", "partition_key" => "key1"},
+             %{"attribute_name" => "value3", "partition_key" => "key1"}
+           ] ==
+             Enum.map(response_data, &ExAws.Dynamo.Decoder.decode/1)
+
+    assert_receive {:span,
+                    span(
+                      name: "DynamoDB_20120810.Query",
+                      kind: :client,
+                      attributes: attributes
+                    )}
+
+    assert %{
+             "aws.dynamodb.attributes_to_get": "null",
+             "aws.dynamodb.consistent_read": "true",
+             "aws.dynamodb.consumed_capacity": "null",
+             "aws.dynamodb.index_name": "null",
+             "aws.dynamodb.limit": "2",
+             "aws.dynamodb.projection": "null",
+             "aws.dynamodb.scan_forward": "true",
+             "aws.dynamodb.table_names": "[\"test_table\"]",
+             "db.system": "dynamodb",
+             "rpc.method": "Query",
+             "rpc.service": "DynamoDB_20120810",
+             "rpc.system": "aws-api"
+           } == :otel_attributes.map(attributes)
+  end
+
+  test "Scan" do
+    create_test_table!()
+
+    data = [
+      %{"partition_key" => "key1", "attribute_name" => "value1", "other_attribute" => "1"},
+      %{"partition_key" => "key1", "attribute_name" => "value3", "other_attribute" => "1"},
+      %{"partition_key" => "key2", "attribute_name" => "value2", "other_attribute" => "1"}
+    ]
+
+    {:ok, _} =
+      ExAws.Dynamo.batch_write_item(%{
+        @table_name => Enum.map(data, &[{:put_request, [{:item, &1}]}])
+      })
+      |> ExAws.request()
+
+    assert {:ok, %{"Items" => response_data}} =
+             ExAws.Dynamo.scan(
+               @table_name,
+               consistent_read: true,
+               filter_expression: "other_attribute= :other_attribute",
+               expression_attribute_values: [other_attribute: "1"],
+               projection_expression: "partition_key, attribute_name",
+               scan_index_forward: false,
+               limit: 1
+             )
+             |> ExAws.request()
+
+    assert [
+             %{"attribute_name" => "value2", "partition_key" => "key2"}
+           ] ==
+             Enum.map(response_data, &ExAws.Dynamo.Decoder.decode/1)
+
+    assert_receive {:span,
+                    span(
+                      name: "DynamoDB_20120810.Scan",
+                      kind: :client,
+                      attributes: attributes
+                    )}
+
+    assert %{
+             "aws.dynamodb.attributes_to_get": "null",
+             "aws.dynamodb.consistent_read": "true",
+             "aws.dynamodb.consumed_capacity": "null",
```

Comment:
There a few attributes that are always null, i could look into this more. Maybe it's the local setup or it might be the telemetry 🤷